junit-jupiter-api/src/main/kotlin/org/junit/jupiter/api/Assertions.kt

```diff
@@ -175,8 +344,15 @@ inline fun <reified T : Throwable> assertThrows(
  * @see Assertions.assertDoesNotThrow
  * @param R the result type of the [executable]
  */
+@OptIn(ExperimentalContracts::class)
 @API(status = STABLE, since = "5.11")
```

Comment:
From the docs: *If a declaration is annotated with [OptIn](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-opt-in/index.html), its usages are not required to opt in to that API.*

It should be safe to add contracts to the methods marked as `STABLE`.